src/commands.ts

```diff
@@ -781,4 +784,136 @@ export async function getVersion() {
+
+    constructor(alias: string, cloudId: string, sk: string, address: string, svg: string, url: string) {
+        // this.balance = 0;
+        this.alias = alias
+        this.cloudId = cloudId;
+        this.sk = sk;
+        this.address = address;
+        this.svg = svg;
+        this.url = url;
+    }
+}
+
+export async function generateGiftCards(prefix: string, quantity: string, cardBalance: string, authToken: string) {
+
+    this.pause();
+    const cloudUrl = process.env.CLOUD_API_ENDPOINT;
+    console.log("cloudUrl = ", cloudUrl)
+    const [total] = await this.account.getShieldedBalances(true);
+    const requiredTotalSum = this.account.humanToShielded(cardBalance) * BigInt(quantity);
+
+    if (requiredTotalSum > total) {
+        this.echo(`total card balance ${requiredTotalSum} exceeds available funds ${this.account.shieldedToHuman(total)}`)
+        return
+    }
+
+    const headers = new Headers();
+    headers.append("Authorization", `Bearer ${authToken}`);
+    headers.append("Content-Type", "application/json");
+    let giftCards: GiftCard[] = [];
+    const treeIndex = (await this.account.getPoolTreeState()).index
+    try {
+        for (let cardIndex = 0; cardIndex < Number(quantity); cardIndex++) {
+            const alias = `${prefix}_${cardIndex}`;
+            const body = JSON.stringify({ "description": `${alias}` });
+            const signupResponse = await fetch(`${cloudUrl}/signup`, {
+                method: 'POST',
+                headers,
+                body
+            });
+            if (signupResponse.status == 401) {
+                throw new Error("not authorized to create new accounts, check admin token in environment variables")
+            } else if (!signupResponse.ok) {
+                throw new Error(`cloud wallet returned bad response ${signupResponse}` )
+            }
+            const signupResponseJson = await signupResponse.json();
+            const cloudId = signupResponseJson.accountId;
+
+            if(!cloudId) throw new Error("sign up response is invalid")
+    
+            const exportResponse = await fetch(`${cloudUrl}/export?id=${cloudId}`);
+
+            if (!exportResponse.ok) throw new Error(`export failed ${exportResponse}`)
+
+            const exportJson = await exportResponse.json();
+            let sk = `0x${exportJson.sk}`;
+    
+            const generateAddressResponse = await fetch(`${cloudUrl}/generateAddress?id=${cloudId}`);
+
+            if (!generateAddressResponse.ok) throw new Error(`generate address failed ${exportResponse}`);
+            const generateAddressResponseJson = await generateAddressResponse.json();
+            const address = generateAddressResponseJson.address;
+            console.log(`generated new account with address: ${address} `);
+    
+            const url = redemptionUrl(sk, treeIndex);
+            const svg = qrcode(url);
+            giftCards.push(new GiftCard(alias, cloudId, sk, address, svg, url));
+        }    
```

Comment:
```suggestion
            giftCards.push(new GiftCard(alias, cloudId, sk, address, svg, url));
            if (Number(quantity) > 1) {
                this.update(-1, `Generating accounts...[${cardIndex + 1}/${Number(quantity)}]`);
            }
        }
        this.update(-1, `Generating account${Number(quantity) > 1 ? 's' : ''}...[[;green;]OK]`);
```

Let's add the simplest progress here. Generating big amount of accounts may take a lot of time